molgenis-data/src/main/java/org/molgenis/data/support/DefaultEntityMetaData.java

```diff
@@ -69,132 +81,211 @@ public DefaultEntityMetaData(String newName, EntityMetaData entityMetaData)
 	 */
 	public DefaultEntityMetaData(EntityMetaData entityMetaData)
 	{
-		this.simpleName = entityMetaData.getSimpleName();
-		this.pack = entityMetaData.getPackage();
+		this(entityMetaData.getName(), entityMetaData);
+	}
+
+	public DefaultEntityMetaData(String simpleName, EntityMetaData entityMetaData)
+	{
+		this.simpleName = simpleName;
 		this.entityClass = entityMetaData.getEntityClass();
+		setPackage(entityMetaData.getPackage());
 		this.label = entityMetaData.getLabel();
 		this.abstract_ = entityMetaData.isAbstract();
 		this.description = entityMetaData.getDescription();
 		EntityMetaData extends_ = entityMetaData.getExtends();
 		this.extends_ = extends_ != null ? new DefaultEntityMetaData(extends_) : null;
 		this.backend = entityMetaData.getBackend();
+		AttributeMetaData idAttr = entityMetaData.getIdAttribute();
+		this.idAttrName = idAttr != null ? idAttr.getName() : null;
+		AttributeMetaData labelAttr = entityMetaData.getLabelAttribute();
+		this.labelAttrName = labelAttr != null ? labelAttr.getName() : null;
+		this.attributes = new CaseInsensitiveLinkedHashMap<>();
+
```

Comment:
Do you mean newline? I put it there on purpose to have a visual separation between entity meta properties and other fields.